### PR TITLE
enhancement: make left sidebar of datadoc resizable

### PR DESCRIPTION
**Pull request by adamstruck**
The DataDoc sidebar should always be resizable.

Prior to this the sidebar was not resizable when table info was pinned. 

https://github.com/pinterest/querybook/assets/5560701/6180b4f4-044f-4915-b271-cb3785a893f5

### CODE REVIEW

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx [hunk 1]
```diff
@@ -143,7 +148,18 @@ export const DataDocLeftSidebar: React.FunctionComponent<IProps> = ({
...
+            ) : (
+                <Resizable
+                    defaultSize={{ width: `${sidebarWidth}px` }}
+                    minWidth={sidebarWidth}
+                    enable={enableResizable({ right: true })}
+                    onResize={resizeToCollapseSidebar}
+                >
+                    {contentDOM}
+                </Resizable>
+            )}
```
Comment (jczhong84) [inline querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:162]: I think we only need either this ` ? :` check or different `sidebarWidth`.  We can keep this.
Try to remove the `HIDDEN_SIDEBAR_WIDTH` and the `useEffect` logic above. Here will be 
```
                <Resizable
                    defaultSize={{ width: `${DEFAULT_SIDEBAR_WIDTH}px` }}
                    minWidth={DEFAULT_SIDEBAR_WIDTH}
                    enable={enableResizable({ right: true })}
                    onResize={resizeToCollapseSidebar}
                >
                    {contentDOM}
                </Resizable>
```

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx [hunk 2]
```diff
@@ -143,7 +148,18 @@ export const DataDocLeftSidebar: React.FunctionComponent<IProps> = ({
                 hidden: cells.length === 0,
             })}
         >
-            {contentDOM}
+            {contentState === 'default' ? (
+                <> {contentDOM} </>
```
Comment (jczhong84) [inline querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:152]: nit: could be just `contentDOM` instead of `<> {contentDOM} </>`

**Comment by jczhong84** [comment]
@adamstruck thanks for your PR! could you please attach some video record if possible next time, which will greatly help the code review, thanks!

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx [hunk 3]
```diff
@@ -76,43 +79,45 @@ export const DataDocLeftSidebar: React.FunctionComponent<IProps> = ({
...
-        React.useCallback(() => setContentState('default'), [])
+        React.useCallback(() => {
+            clearSidebarTableId();
+            setContentState('default');
+        }, [])
     );
 
+    useEffect(() => {
+        if (contentState === 'default') {
+            setSidebarWidth(HIDDEN_SIDEBAR_WIDTH);
```
[the remark this answers is not in the document]
Comment (adamstruck) [reply querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:90]: I removed `HIDDEN_SIDEBAR_WIDTH`.

**Comment by adamstruck** [comment]
@jczhong84 I pushed a commit addressing your comments and added a screen recording.

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss [hunk 4]
```diff
@@ -26,9 +26,7 @@
         width: 100%;
         height: calc(100vh - 158px);
         max-height: 1200px;
-    }
 
-    .sidebar-content-table {
```
Comment (jczhong84) [inline querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss:31]: why remove this?

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx [hunk 5]
```diff
@@ -57,13 +57,11 @@ export const DataDocLeftSidebar: React.FunctionComponent<IProps> = ({
...
-        () => () => {
-            clearSidebarTableId();
-            setContentState('default');
-        },
-        []
-    );
+    useEffect(() => {
+        clearSidebarTableId();
+        setContentState('default');
+    }, []);
```
Comment (jczhong84) [inline querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:63]: why this change?

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss (hunk 4 above)
Comment (adamstruck) [reply querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss:31]: Seemed redundant since the styles above also applied to the same class. 

Can revert.

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx [hunk 6]
```diff
@@ -57,13 +57,11 @@ export const DataDocLeftSidebar: React.FunctionComponent<IProps> = ({
         }, [])
     );
 
-    useEffect(
-        () => () => {
-            clearSidebarTableId();
```
Comment (adamstruck) [inline querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:62]: Opportunistic cleanup - removes the extra arrow function `() => () => {`

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx (hunk 5 above)
Comment (adamstruck) [reply querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:63]: Opportunistic cleanup - removes the extra arrow function `() => () => {`

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss (hunk 4 above)
Comment (jczhong84) [reply querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.scss:31]: Looks like the above css applies to both table and content, but this `.sidebar-content-table` only applies to the table. Let's revert this change in case, unless you've fully tested it and it looks good.

File: querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx (hunk 5 above)
Comment (jczhong84) [reply querybook/webapp/components/DataDocLeftSidebar/DataDocLeftSidebar.tsx:63]: `()=>()=>{}`  will do the cleanup when the components unmounts, while `()=>{}` cleans up when the components mounts.

just to make sure if the ui behavior still looks good

**Review by jczhong84 — APPROVED** [review]

**Comment by jczhong84** [comment]
thanks for the updates
